### PR TITLE
[ServiceBus] mode  (ReceiveMode) parameter needs better exception behavior

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
If it is mis-typed now it will return an AttributeError further down the stack without useful guidance (and only does so when creating the handler, as well)

Will now return a TypeError at initialization.

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/receiver_mixins.py [hunk 1]
```diff
@@ -33,6 +33,10 @@ def _populate_attributes(self, **kwargs):
         self._auth_uri = "sb://{}/{}".format(self.fully_qualified_namespace, self.entity_path)
         self._entity_uri = "amqps://{}/{}".format(self.fully_qualified_namespace, self.entity_path)
         self._mode = kwargs.get("mode", ReceiveSettleMode.PeekLock)
+        # While we try to leave failures to the service, in this case the errors lower down the stack are less clear.
+        if not isinstance(self._mode, ReceiveSettleMode):
+            raise TypeError("Parameter 'mode' must be of type ReceiveSettleMode")
+
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/receiver_mixins.py:39]: Assume the text will be "ReceiveMode" after consistency PR is merged.

**Review by yunhaoling — DISMISSED** [review]

**Comment by yunhaoling** [comment]
do we want to mention this in changelog?

**Comment by KieranBrantnerMagee** [comment]
Good callout as always Adam; even though I had waved it off mentally as "it's still an exception" we should be precise.  Since tests are green and that's only a doc change I'll likely push it through after with a rubber stamp.

**Review by yunhaoling — APPROVED** [review]

**Pull request merged**